components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -330,20 +330,15 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
         // Only exception is when we do the switch from the "old" to the "new" one.
         const startedInstanceId = this.state?.startedInstanceId;
         if (startedInstanceId !== workspace.status.instanceId) {
-            // do we want to switch to "new" instance we just received an update for?
```

Comment:
@mustard-mh How do we now guard against this problem? :thinking: 

> Here we filter out updates to instances we haven't started to avoid issues with updates coming in out-of-order
        // (e.g., multiple "stopped" events from the older instance, where we already started a fresh one after the first)

Reply:
Pushed a change to this file: call `getWorkspace` first to get the really latest instanceId, could you take a look?

Reply:
statusVersion is depends on the activity time, so if the situation will happen, we will send it anyway with newer statusVersion
> // (e.g., multiple "stopped" events from the older instance, where we already started a fresh one after the first)

Reply:
cc @geropl

Reply:
> Pushed a change to this file: call getWorkspace first to get the really latest instanceId, could you take a look?

Sorry I'm not seeing it. Can you share a link maybe?

Reply:
here you are https://github.com/gitpod-io/gitpod/pull/19748/commits/23a199b65b2254ff75f3f0fcac285b4b94ff31a4

Reply:
`getWorkspace` will get first instance from DB (order by creationTime DESC)

Reply:
Ah, nice! Yeah, IMO that's the best solution here. :100: